astropy/utils/iers/iers.py

```diff
@@ -442,6 +442,36 @@ class because it has different requirements.
                 warn(msg, IERSDegradedAccuracyWarning)
             # No IERS data covering the time(s) and user is OK with no warning.
 
+    def _lagrange_interp(self, x_vals, y_vals, xint):
+        val = np.zeros((y_vals.shape[1:]), dtype=y_vals.dtype)
+
+        for i in range(len(x_vals)):
+            term = y_vals[i]
+            for j in range(len(x_vals)):
+                if i != j:
+                    diff = x_vals[i] - x_vals[j]
+                    diff = np.where(diff == 0, 1e-10, diff)
+                    term *= (xint - x_vals[j]) / diff
+            val += term
+
+        return val
+
+    def _ut1_utc_interp(self, i0, i1, mjd):
+        indices = np.array([i0 - 1, i0, i1, i1 + 1])
+        indices = np.clip(indices, 0, len(self) - 1)
```

Comment:
I think you'll need to set up the clipping of `i1` on l.462 (pre-change) in `interpolate` differently instead of redoing the clipping here. Note that if you look at https://hpiers.obspm.fr/iers/models/interp.f, it is not the indices that are clipped - rather it is made sure that there are always three values.